### PR TITLE
Run phpcs through eslines

### PR DESCRIPTION
**Pull request by mjangda**
To help us incrementally apply coding standards to the codebase.

This adds package.json and a new script (npm run phpcs) that can be run to pass only changed code against PHPCS. Circle CI is wired up to run it when the RUN_EXTRA env var is set.

To test, I added an intentional error in fbdd94a which was then caught in https://circleci.com/gh/Automattic/vip-go-mu-plugins/2914

```
> @automattic/vip-go-mu-plugins@1.0.0 phpcs:changed /home/circleci/project
> vendor/bin/phpcs --report=bin/phpcs-eslint-report.php | eslines --quiet

/home/circleci/project/000-vip-init.php
  12:1  error  error_log() found. Debug code should not normally be used in production  WordPress.PHP.DevelopmentFunctions.error_log_error_log

✖ 1 problem (1 error, 0 warnings)
```

The error was removed in cba55b0 and now the CI build passes (https://circleci.com/gh/Automattic/vip-go-mu-plugins/2918):

```
> @automattic/vip-go-mu-plugins@1.0.0 phpcs:changed:remote /home/circleci/project
> vendor/bin/phpcs --report=bin/phpcs-eslint-report.php | eslines --quiet

<no output means pass>
```

### CODE REVIEW

File: phpcs.xml.dist [hunk 1]
```diff
@@ -55,11 +56,12 @@
...
+	<rule ref="WordPress-Core" />
+	<rule ref="WordPress-Extra" />
+
 	<!-- Rules: Check VIP Coding Standards - see
 		https://github.com/Automattic/VIP-Coding-Standards/ -->
 	<rule ref="WordPress-VIP-Go"/>
 	
-	<!-- Rules: Check WordPress Coding Standards -->
-	<rule ref="WordPress-Core"/>
-	<rule ref="WordPress-Docs"/>
```
Comment (mjangda) [inline phpcs.xml.dist]: I removed Docs because it's a little too prescriptive for our needs.

File: phpcs.xml.dist [hunk 2]
```diff
@@ -55,11 +56,12 @@
 		https://github.com/PHPCompatibility/PHPCompatibility#sniffing-your-code-for-compatibility-with-specific-php-versions -->
 	<config name="testVersion" value="7.3-"/>
 
+	<!-- Rules: Check WordPress Coding Standards -->
+	<rule ref="WordPress-Core" />
+	<rule ref="WordPress-Extra" />
```
Comment (mjangda) [inline phpcs.xml.dist]: I added Extra as it adds a nice set of best practices to follow.

File: phpcs.xml.dist [hunk 3]
```diff
@@ -28,17 +28,18 @@
...
+	<exclude-pattern>elasticsearch/debug-bar-elasticpress/*</exclude-pattern>
 	<exclude-pattern>shared-plugins/*</exclude-pattern>
 	<exclude-pattern>vaultpress/*</exclude-pattern>
 	<exclude-pattern>vendor/*</exclude-pattern>
 	<exclude-pattern>wordpress-importer/*</exclude-pattern>
 	<exclude-pattern>wp-cron-control/*</exclude-pattern>
 
 	<!-- How to scan -->
-	<!-- Show sniff and progress -->
-	<arg value="sp"/>
```
Comment (mjangda) [inline phpcs.xml.dist]: This was removed to allow piping the results in our `npm` script.

File: composer.json [hunk 4]
```diff
@@ -3,7 +3,6 @@
         "phpunit/phpunit": "~6.5",
         "automattic/vipwpcs": "^2.0",
         "phpcompatibility/phpcompatibility-wp": "^2.1",
-        "dealerdirect/phpcodesniffer-composer-installer": "^0.5.0",
-        "exussum12/coverage-checker": "^0.11.2"
```
Comment (mjangda) [inline composer.json]: No longer used.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "@automattic/vip-go-mu-plugins",
+  "private": true,
+  "version": "1.0.0",
+  "description": "mu-plugins used on the VIP Go platform",
+  "scripts": {
+    "phpcs": "vendor/bin/phpcs",
+    "phpcs:changed:remote": "vendor/bin/phpcs --report=bin/phpcs-eslint-report.php | eslines --quiet",
+    "phpcs:changed:local": "vendor/bin/phpcs --report=bin/phpcs-eslint-report.php | eslines --quiet -d index"
```
Comment (nickdaugherty) [inline package.json]: What's the difference between `phpcs:changed:remote` and `:local`?

Comment (mjangda) [reply package.json]: > --diff or -d: let you choose between two diff strategies for the downgrade-unmodified-lines processor
> index: to diff HEAD against the git index.
> remote: to diff HEAD against the git remote. This is the default.

In our case, index => local and remote => remote.

We'd use `:local` for a pre-commit hook, and `:remote` for PRs / CI.

https://github.com/Automattic/eslines#runtime-options

**Comment by mjangda** [comment]
No deploy needed here.

**Pull request merged**